kubernetes-sigs/vsphere-csi-driver
# Volume health status [Inaccessible] test cases

**What this PR does / why we need it**: This PR has code changes to adapt latest govmomi release and also making the Volume Health Status Inaccessible scenarios more robust

Test cases modified:

1. Verify health annotation is updated from unknown status to inaccessible
2. Verify health annotation is not updated to unknown status from inaccessible
3. Verify health annotation added on the pvc is changed from accessible to inaccessible
4. Verify Volume health Inaccessible on Statefulset
5. Verify Inaccesssible Volume health when GC CSI is down
6. Verify health status of pvc after bringing SV API server down

E2E Execution logs:
[accessible_inaccessible.txt](https://github.com/kubernetes-sigs/vsphere-csi-driver/files/5785613/accessible_inaccessible.txt)
[from_unknown_inaccessible.txt](https://github.com/kubernetes-sigs/vsphere-csi-driver/files/5785614/from_unknown_inaccessible.txt)
[gc_statefulset.txt](https://github.com/kubernetes-sigs/vsphere-csi-driver/files/5785615/gc_statefulset.txt)
[sv_api_2.txt](https://github.com/kubernetes-sigs/vsphere-csi-driver/files/5785617/sv_api_2.txt)
[unknown_status_inacc.txt](https://github.com/kubernetes-sigs/vsphere-csi-driver/files/5785618/unknown_status_inacc.txt)

Execution logs
https://gist.github.com/marunachalam/3acdff8038f1ed0dcf645c87954b9b13
